### PR TITLE
Create the Why Elixir blogpost

### PR DESCRIPTION
**Pull request by mariari**

### CODE REVIEW

File: mariari/why-elixir.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
+
+# Table of Contents
```
Comment (0xapriori) [inline mariari/why-elixir.md]: You can remove TOC. A TOC will autopopulate on the left side of the article. Before "Anoma: Why Elixir?" you need an excerpt. The excerpt will appear underneath the title and author information but above the media. See this post as an example. 

https://anoma.net/research/superspartan-by-hand

here is the syntax you want:

---
title: 
publish_date: 
category: 
image: 
excerpt: 

---

File: mariari/why-elixir.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    4.  [Conclusion](#org8571bee)
+
+
+
+<a id="org9b02694"></a>
+
+# Anoma: Why Elixir?
+
+Anoma is a decentralized operating system with heterogeneous
+trust. For developing Anoma we've We've been using [Elixir](https://elixir-lang.org/). This choice
```
Comment (0xapriori) [inline mariari/why-elixir.md]:
```suggestion
trust. For developing Anoma we've been using [Elixir](https://elixir-lang.org/). This choice
```

File: mariari/why-elixir.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+is somewhat unconventional, and thus begs the question "why write
+Anoma in Elixir!?". There are a plethora of reasons as to why [Elixir](https://elixir-lang.org/)
+is a good choice, and to motivate the reasoning we will go first from
```
Comment (0xapriori) [inline mariari/why-elixir.md]:
```suggestion
is a good choice. To motivate the reasoning, we will go first from
```

File: mariari/why-elixir.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+## Live Interactive Systems
+
+When building any large complicated system, the number one feature one
+wants is to be able to understand how the system is operating along
+with being able to determine if there are mishaps when carrying out a
+design. For most software projects, this means sitting down and
+thinking about the system abstractly or tracing through the system
+execution in the case of direct questions. Thankfully for us, we have
+chosen [Elixir](https://elixir-lang.org/), which rest on the foundations of Erlang/OTP. In
```
Comment (0xapriori) [inline mariari/why-elixir.md]:
```suggestion
chosen [Elixir](https://elixir-lang.org/), which rests on the foundations of Erlang/OTP. In
```

File: mariari/why-elixir.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+which we can ask the system questions.
+
+For example, instead of having to sit and think about the
+architecture of how the components of Anoma fit together, I can ask
+the underlying system, "please give me the layout of processes for my
+application":
+
+![img](https://pomf2.lain.la/f/wqg712cm.png)
+
+Knowing the specifics of this diagram doesn't matter, however what
```
Comment (0xapriori) [inline mariari/why-elixir.md]:
```suggestion
Knowing the specifics of this diagram doesn't matter. However what
```

File: mariari/why-elixir.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+does matter is that we can invite the system we are building into help
```
Comment (0xapriori) [inline mariari/why-elixir.md]: i think I know what you are trying to say - the system can help explain itself. The word choice here is a bit difficult to parse.

File: mariari/why-elixir.md [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+explaining itself. And if we have further questions about the system,
+we can drill down and ask it more questions. For example what if we
+was curious about one of those nodes in the diagram, what information
```
Comment (0xapriori) [inline mariari/why-elixir.md]:
```suggestion
were curious about one of those nodes in the diagram, what information
```

File: mariari/why-elixir.md [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+can Ι get from there?:
```
Comment (0xapriori) [inline mariari/why-elixir.md]:
```suggestion
can we get from there?:
```

File: mariari/why-elixir.md [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+        through the language itself, proving itself capable enough to
+        compose a highly robust system.
+
+With these properties and features laid out, this make Elixir quite
+different from language in common use (Java, Rust, Python etc.)
+
+With that said, there are [genres of languages](https://wiki.c2.com/?ImageBasedLanguage) that exhibit the same
+features some of which have [superior visualization techniques](https://gtoolkit.com/). What is
+it about the Erlang/OTP system that makes them special compared to
+them?
```
Comment (0xapriori) [inline mariari/why-elixir.md:123]: not sure what the second "them" is referring to - other languages?

File: mariari/why-elixir.md [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+        processes (some of who may be supervisors). When a process they
+        are supervising crashes, they have policies on what happens to
+        the other processes under their control.
+    -   It was realized that actors on their own are not enough to make a
+        robust system, and this is a crucial component that is missing
+        from most other actor models.
+5.  A let it crash mentality is had
+    -   With Supervisors this is a very potent strategy for creating
+        reliable and robust systems.
+        -   When crashes early happens, we can isolate the errors, this is
```
Comment (0xapriori) [inline mariari/why-elixir.md]:
```suggestion
        -   When crashes happen early, we can isolate the errors, this is
```

File: mariari/why-elixir.md [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+    of this at play on their real systems!
+4.  Processes are managed by a chain of [supervisors](https://www.erlang.org/doc/apps/stdlib/supervisor.html).
+    -   Supervisors are interesting in that they look after many
+        processes (some of who may be supervisors). When a process they
+        are supervising crashes, they have policies on what happens to
+        the other processes under their control.
+    -   It was realized that actors on their own are not enough to make a
+        robust system, and this is a crucial component that is missing
+        from most other actor models.
+5.  A let it crash mentality is had
```
Comment (0xapriori) [inline mariari/why-elixir.md]: had?

**Review by 0xapriori — COMMENTED** [review]
Overall, very good high level article. 

Aside from adding the excerpt, I think it maybe helpful to add either an introduction or a bulleted tl;dr as to what the article will discuss; e.g. I just opened this link because Adrian shared it on twitter, why am I reading this article. 

example:

>The article explains why Anoma is being built using the Elixir programming language:

>- Elixir, which runs on the Erlang virtual machine, lets developers inspect and debug the system while it's running and update the code without restarting
>- Erlang was created to build robust, fault-tolerant networked systems using lightweight processes that can crash without taking down the whole system
>- Elixir's process model is a natural fit for Anoma's architecture, which is based around separate "engines" that communicate by passing messages 
>- Elixir makes it easier to implement Anoma's design with less accidental complexity, because the language matches the architecture
>- Having the implementation closely match the specification will make it easier to mathematically prove things about the properties of the system down the road

> In short, Elixir provides the right tools and abstractions to build Anoma according to spec, while providing a good developer experience and supporting future verification efforts.

One other thing that could use improvement is transition between sections. Perhaps one scentence summary of what we just talked about and where we are going next. 

In the components of Anoma Node in the final concluding paragraph it may make sense to breifly explain the node architecture in advance with some sort of diagram. Perhpas out of scope.

File: mariari/why-elixir.md [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+properties can fundamentally depend on your choice of a language.
+
+For example, if you are interested in functions, it matters whether
+you present id : Bool -> Bool as lambda x -> x or {{true, true},
+{false, false}}. There is a reason why we don't use set theory
+everywhere in mathematics. If you think of functions as something that
+takes things in and pops things out, the type theoretic approach is
+probably easier to comprehend and work with.
+
+Folliwing this idea, a good requirement for the desired language is to
```
Comment (agureev) [inline mariari/why-elixir.md]:
```suggestion
Following this idea, a good requirement for the desired language is to
```

File: mariari/why-elixir.md [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+makes it easy to establish a connection between concepts central to
+the specification and our code. Otherwise, the application becomes
+hard to adopt and develop for any outside party. Having such
+properties can fundamentally depend on your choice of a language.
+
+For example, if you are interested in functions, it matters whether
+you present id : Bool -> Bool as lambda x -> x or {{true, true},
+{false, false}}. There is a reason why we don't use set theory
+everywhere in mathematics. If you think of functions as something that
+takes things in and pops things out, the type theoretic approach is
```
Comment (agureev) [inline mariari/why-elixir.md:244]:
```suggestion
take things in and pops things out, the type theoretic approach is
```

File: mariari/why-elixir.md [hunk 14]
```diff
@@ -0,0 +1,277 @@
...
+probably easier to comprehend and work with.
+
+Folliwing this idea, a good requirement for the desired language is to
+have a good first-class citizen corresponding to the central concept
+of the specification: the [engine](https://specs.anoma.net/latest/node_architecture/engines.html?h=engine). Almost every functionality in the
+Anoma infrastructure is specified as a functionality of a particular
+engine. Mempool functionality is done through the Mempool
+Engine. Transaction execition is done through the Execution Engine.
```
Comment (agureev) [inline mariari/why-elixir.md]:
```suggestion
Engine. Transaction execution is done through the Execution Engine.
```

File: mariari/why-elixir.md [hunk 15]
```diff
@@ -0,0 +1,277 @@
...
+{false, false}}. There is a reason why we don't use set theory
+everywhere in mathematics. If you think of functions as something that
+takes things in and pops things out, the type theoretic approach is
+probably easier to comprehend and work with.
+
+Folliwing this idea, a good requirement for the desired language is to
+have a good first-class citizen corresponding to the central concept
+of the specification: the [engine](https://specs.anoma.net/latest/node_architecture/engines.html?h=engine). Almost every functionality in the
+Anoma infrastructure is specified as a functionality of a particular
+engine. Mempool functionality is done through the Mempool
```
Comment (agureev) [inline mariari/why-elixir.md]:
```suggestion
engine. Mempool behavior is implemented through the Mempool
```

File: mariari/why-elixir.md (hunk 13 above)
Comment (mariari) [reply mariari/why-elixir.md:244]: takes is correct

File: mariari/why-elixir.md (hunk 9 above)
Comment (mariari) [reply mariari/why-elixir.md:123]: yeah those other languages

File: mariari/why-elixir.md (hunk 11 above)
Comment (mariari) [reply mariari/why-elixir.md]: Rewrote the sentence to remove the had bit

File: mariari/why-elixir.md (hunk 6 above)
Comment (mariari) [reply mariari/why-elixir.md]: Rewrote it to say this instead:


The specifics of the diagram doesn't matter. What does matter is that
the system can reflect on itself, helping to explain the very system
that we are building. Meaning, that the system we are building can be
in dialog with the builders to shape discussion around itself. No
longer is the system simply speculated about from the hazy
recollections that the engineers who made it have. A great upshot of
this is that if we have further questions about the system, we can
drill down and ask it more questions. For example what if we were
curious about one of those nodes in the diagram, what information can
we get from there?:

File: mariari/why-elixir.md [hunk 16]
```diff
@@ -0,0 +1,308 @@
+---
+title: Why Anoma
+publish_date:
```
Comment (0xapriori) [inline mariari/why-elixir.md]: - Can add a publish date - can be today / tomorrow. Won't update on the blog until the PR is merged. 
- For image you want to create a media folder then reference the item in image. I'm not sure what you have in mind. Typically, I use one of the AI image generating services to get image that gestures at the article. 
- For excerpt: Anoma is a decentralized operating system with heterogeneous
trust. For developing Anoma we've been using [Elixir](https://elixir-lang.org/). This choice is somewhat unconventional, and thus begs the question "why write Anoma in Elixir!?"
- For Category the choices are: compilers, distributed-systems, ecosystem, community, cryptography, applications, security, community

**Review by 0xapriori — APPROVED** [review]

**Pull request merged**